collect/panic_handler.go

```diff
@@ -0,0 +1,16 @@
+package collect
+
+import (
+	"fmt"
+	"runtime/debug"
+)
+
+func catchPanic(fn func()) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic in central collector: %v\n%s", r, debug.Stack())
```

Comment:
I think we probably want to log this to the injected logger.

Reply:
it's logged on the caller side